### PR TITLE
CON-311: Add warning to documentation

### PR DESCRIPTION
**Pull request by Alxe**
See #198 for more context.

### CODE REVIEW

**Comment by Alxe** [comment]
@lloadholt Can you review the wording?

File: docs/release_notes.rst [hunk 1]
```diff
@@ -9,6 +9,11 @@ Supported Platforms
   * v18
   * v20
 
+.. warning::
+  Due to an upstream change in Node.js, **some versions will fail when building
```
Comment (lloadholt) [inline docs/release_notes.rst:13]: @Alxe Is "upstream change in Node.js" meaningful for customers? If so, what changed and why and when? If not, can this be removed?

Comment (Alxe) [reply docs/release_notes.rst:13]: It's meaningful, but I'm not sure how much information we'd need to provide. 

1. We're using [`node-ffi-napi`](https://github.com/node-ffi-napi/ref-napi/) as our dependency. 
1. Node [21.6](https://nodejs.org/en/blog/release/v11.6.0) initially broke this package by adding [this PR](https://github.com/nodejs/node/pull/50060). We didn't explicitly support nor test this release.
1. The same breaking change was then ported to [18.20](https://nodejs.org/en/blog/release/v18.20.0) and [20.12](https://nodejs.org/en/blog/release/v20.12.0). These were our target releases, which is why we're affected now.

An alternative approach is adding this context in the Github issue that I created, #198, and linking to that issue from the documentation.

Comment (Alxe) [reply docs/release_notes.rst:13]: The upstream change was made, for whatever the reason, and this broke a package that RTI depended on.
It seems that there's an on-going discussion on whether this upstream change will be reverted at some point, see [this](https://github.com/nodejs/node/issues/52229).

I think adding it here would be enough, the Supported Platforms and Release notes are in the same page and a big Warning should be enough. The known issue is also documented in #198 for anyone not taking a look at the actual documentation.

Comment (lloadholt) [reply docs/release_notes.rst:13]: OK. My only suggestion is to remove the :code: format on the version numbers to be consistent with the rest of the page.

Comment (Alxe) [reply docs/release_notes.rst:13]: Done in latest commit.

File: docs/release_notes.rst [hunk 2]
```diff
@@ -5,10 +5,15 @@ Supported Platforms
 -------------------
 
 *Connector* has been validated to run with the following Node.js versions:
-  * v17
   * v18
+  * v19
```
Comment (Alxe) [inline docs/release_notes.rst:9]: I updated this based on our CI tests. I don't know why we said v17, when we're only testing v18.
While v17 is likely to work, it's EOL and not something we've explicitly validated.

**Review by lloadholt — APPROVED** [review]

**Review by alexcamposruiz — APPROVED** [review]

File: docs/release_notes.rst [hunk 3]
```diff
@@ -5,10 +5,16 @@ Supported Platforms
...
   * v20
 
+.. warning::
+  Due to an upstream change in Node.js, **some versions will fail when building
+  Connector dependencies**. This issue affects, at the very least,
+  Node.js versions 18.20, 20.12 and 21.6.
+  This is tracked in `Github Issue #198 <https://github.com/rticommunity/rticonnextdds-connector-js/issues/198>`_.
```
Comment (lloadholt) [inline docs/release_notes.rst:16]:
```suggestion
  For more information, see `Github Issue #198 <https://github.com/rticommunity/rticonnextdds-connector-js/issues/198>`_.
```

**Review by elainerti — APPROVED** [review]

**Pull request merged**